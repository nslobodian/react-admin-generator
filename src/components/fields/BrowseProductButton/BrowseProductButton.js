import React, { Component } from 'react'
import s from './BrowseProductButton.scss'
import connect from 'react-redux/es/connect/connect'
import { show } from 'redux-modal/lib/actions'
import cn from 'classnames'
import SelectProductModal from '../../modals/SelectProductModal'
import { clearProduct, saveProduct } from 'redux/Products'

export class BrowseProductButton extends Component {
  clearProductHandle = () => {
    this.props.input.onChange(null)
    this.props.clearProduct(this.props.formName)
  }

  saveProductHandler = ({ productId, mainPictureUrl, productUrl }) => {
    const { saveProduct, input: { onChange }, formName } = this.props
    onChange({ productId, photo: mainPictureUrl })
    saveProduct(formName, productUrl)
  }

  render () {
    const { showProductsModal, className, input: { value }, meta: { touched, invalid, error } } = this.props
    return <div>
      <div className={s.photoTwoButtonsWrapper}>
        <button type='button' className={cn('btn btn-primary', className)} onClick={showProductsModal}>
          Browse
        </button>
        {
          value &&
          <button type='button' className={cn('btn btn-danger', className)} onClick={this.clearProductHandle}>
            Delete</button>
        }
        <SelectProductModal saveProduct={this.saveProductHandler} />
      </div>
      { touched && invalid &&
      <div className='alert alert-danger m-t-1' role='alert'>{error}</div>}
    </div>
  }
}

const mapActionCreators = {
  showProductsModal: () => show('SelectProductModal'),
  clearProduct,
  saveProduct,
}

export default connect(() => ({}), mapActionCreators)(BrowseProductButton)
