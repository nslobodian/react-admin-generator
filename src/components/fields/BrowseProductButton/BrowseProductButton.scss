.photoTwoButtonsWrapper {
  display: flex;

  button {
    width: 100%;
  }
}