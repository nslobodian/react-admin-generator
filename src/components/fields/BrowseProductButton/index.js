import BrowseProductButton from './BrowseProductButton'

export default BrowseProductButton
