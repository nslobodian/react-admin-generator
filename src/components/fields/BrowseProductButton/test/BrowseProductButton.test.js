import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { BrowseProductButton } from '../BrowseProductButton'

const props = {
  input: {
    value: 1,
    onChange: jest.fn(),
  },
    meta: {
      touched: 'touched',
      invalid: 'invalid',
      error: 'error',
    },
    className: 'className',
    showProductsModal: jest.fn(),
    clearProduct: jest.fn(),
    saveProduct: jest.fn(),
    productUrl: 'productUrl',
    formName: 'formName',
}

describe('(Components) BrowseProductButton', () => {
  it('Should render component properly', () => {
    const wrapper = shallow(
      <BrowseProductButton
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.instance().clearProductHandle()
  }),
  it('Should render component properly', () => {
    const productId = 1;
    const mainPictureUrl = 'mainPictureUrl';
    const productUrl = 'mainPictureUrl';
    const wrapper = shallow(
      <BrowseProductButton
        {...props}
      />)
    wrapper.instance().saveProductHandler(productId, mainPictureUrl, productUrl)
  })
})
