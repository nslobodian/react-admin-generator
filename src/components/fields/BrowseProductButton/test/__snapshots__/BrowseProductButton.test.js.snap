// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) BrowseProductButton Should render component properly 1`] = `
<div>
  <div
    className="photoTwoButtonsWrapper"
  >
    <button
      className="btn btn-primary className"
      onClick={[Function]}
      type="button"
    >
      Browse
    </button>
    <button
      className="btn btn-danger className"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
    <Connect(ConnectModal(SelectProductModalWrapper))
      saveProduct={[Function]}
    />
  </div>
  <div
    className="alert alert-danger m-t-1"
    role="alert"
  >
    error
  </div>
</div>
`;
